### ISSUE TITLE
Given exponent overflowing the maximum accepted scale (255).: TryFromIntError(())

### ISSUE THREAD

**Issue by agz-jrugel** [prisma/prisma#14703]
Hi Prisma Team! My Prisma Client just crashed. This is the report:
## Versions

| Name            | Version            |
|-----------------|--------------------|
| Node            | v16.16.0           | 
| OS              | debian-openssl-1.1.x|
| Prisma Client   | 4.1.1              |
| Query Engine    | 8d8414deb360336e4698a65aa45a1fbaf1ce13d8|
| Database        | sqlserver          |

## Logs
```
prisma:tryLoadEnv Environment variables loaded from /home/jrugel/Webs/documentos-electronicos/.env
prisma:tryLoadEnv Environment variables loaded from /home/jrugel/Webs/documentos-electronicos/.env
prisma:client dirname /home/jrugel/Webs/documentos-electronicos/node_modules/.prisma/client
prisma:client relativePath ../../../prisma
prisma:client cwd /home/jrugel/Webs/documentos-electronicos/prisma
prisma:client clientVersion: 4.1.1
prisma:client clientEngineType: library
prisma:client:libraryEngine internalSetup
prisma:client:libraryEngine:loader Searching for Query Engine Library in /home/jrugel/Webs/documentos-electronicos/node_modules/.prisma/client
prisma:client:libraryEngine:loader loadEngine using /home/jrugel/Webs/documentos-electronicos/node_modules/.prisma/client/libquery_engine-debian-openssl-1.1.x.so.node
prisma:tryLoadEnv Environment variables loaded from /home/jrugel/Webs/documentos-electronicos/.env
prisma:client dirname /home/jrugel/Webs/documentos-electronicos/node_modules/.prisma/client
prisma:client relativePath ../../../prisma
prisma:client cwd /home/jrugel/Webs/documentos-electronicos/prisma
prisma:client clientVersion: 4.1.1
prisma:client clientEngineType: library
prisma:client:libraryEngine internalSetup
prisma:client:libraryEngine:loader Searching for Query Engine Library in /home/jrugel/Webs/documentos-electronicos/node_modules/.prisma/client
prisma:client:libraryEngine:loader loadEngine using /home/jrugel/Webs/documentos-electronicos/node_modules/.prisma/client/libquery_engine-debian-openssl-1.1.x.so.node
prisma:tryLoadEnv Environment variables loaded from /home/jrugel/Webs/documentos-electronicos/.env
prisma:client dirname /home/jrugel/Webs/documentos-electronicos/node_modules/.prisma/client
prisma:client relativePath ../../../prisma
prisma:client cwd /home/jrugel/Webs/documentos-electronicos/prisma
prisma:client clientVersion: 4.1.1
prisma:client clientEngineType: library
prisma:client:libraryEngine internalSetup
prisma:client:libraryEngine:loader Searching for Query Engine Library in /home/jrugel/Webs/documentos-electronicos/node_modules/.prisma/client
prisma:client:libraryEngine:loader loadEngine using /home/jrugel/Webs/documentos-electronicos/node_modules/.prisma/client/libquery_engine-debian-openssl-1.1.x.so.node
prisma:client:libraryEngine sending request, this.libraryStarted: false
prisma:client:libraryEngine library starting
prisma:client:libraryEngine library started
```

## Client Snippet
```js
import { PrismaClient } from "@prisma/client";

export default async function handler(req, res) {
  try {
    const prisma = new PrismaClient();
    const des = await prisma.de.findMany({
      include: {
        estado_historico: true,
      },
    });
    console.dir(des, { depth: null });

    res.status(200).json({ ok: true });
  } catch (error) {
    console.error(error);
    res.status(200).json({ error: true });
  }
}
```

## Schema
```prisma
generator client {
  provider = "prisma-client-js"
}

datasource db {
  provider = "sqlserver"
  url      = env("DATABASE_URL")
}

model archivo {
  id      Decimal  @id(map: "pk_archivo") @db.Decimal(19, 0)
  de_id   Decimal  @db.Decimal(19, 0)
  tipo    String   @db.VarChar(3)
  data    String?  @db.Text
  tstamp  DateTime
  impreso String?  @db.Char(1)
  de      de       @relation(fields: [de_id], references: [id], onUpdate: NoAction, map: "fk_archivo_de_id")
}

model autofactura {
  id                              Decimal  @id(map: "pk_autofactura") @db.Decimal(19, 0)
  procesado                       Boolean  @default(false, map: "DF__autofactu__proce__3A81B327")
  de_id                           Decimal? @db.Decimal(19, 0)
  condicion_venta                 Int?
  monto                           Decimal? @db.Decimal(28, 0)
  modalidad_transporte            Int?
  responsable_emision_nota        Int?
  responsable_pago                Int?
  condicion_entrega               String?  @db.VarChar(255)
  naturaleza_vendedor_codigo      Int
  tipo_documento_vendedor_codigo  Int
  numero_documento_vendedor       String   @db.VarChar(20)
  nombre_completo_vendedor        String   @db.VarChar(60)
  direccion_vendedor              String   @db.VarChar(255)
  numero_de_casa_vendedor         Int
  departamento_vendedor_codigo    Int
  distrito_vendedor_codigo        Int?
  ciudad_vendedor_codigo          Int
  lugar_transaccion               String   @db.VarChar(255)
  departamento_transaccion_codigo Int
  distrito_transaccion_codigo     Int
  ciudad_transaccion_codigo       Int
  when_created                    DateTime
  when_modified                   DateTime
  de                              de?      @relation(fields: [de_id], references: [id], onDelete: NoAction, onUpdate: NoAction, map: "fk_autofactura_de_id")
  pago                            pago[]
}

model cuota {
  id                Decimal  @id(map: "pk_cuota") @db.Decimal(19, 0)
  factura_id        Decimal? @db.Decimal(19, 0)
  monto_cuota       Decimal  @db.Decimal(28, 0)
  fecha_vencimiento DateTime @db.Date
  moneda_codigo     String?  @db.VarChar(10)
  factura           factura? @relation(fields: [factura_id], references: [id], onDelete: NoAction, onUpdate: NoAction, map: "fk_cuota_factura_id")
}

model de {
  id                                           Decimal              @id(map: "pk_de") @db.Decimal(19, 0)
  fecha_hora_emision                           DateTime
  tipo_impuesto                                Int
  indicador_presencia_codigo                   Int?
  indicador_presencia_descripcion              String?              @db.VarChar(255)
  tipo_transaccion_codigo                      Int?
  tipo_transaccion_descripcion                 String?              @db.VarChar(255)
  informacion_extra_emisor                     String?              @db.VarChar(255)
  informacion_extra_fisco                      String?              @db.VarChar(255)
  numero_timbrado                              String?              @db.VarChar(255)
  establecimiento_codigo                       Int?
  punto_expedicion_codigo                      Int?
  numero_documento                             Int?
  fecha_inicio_timbrado                        DateTime?            @db.Date
  fecha_fin_timbrado                           DateTime?            @db.Date
  naturaleza_receptor_codigo                   Int?
  tipo_operacion_codigo                        Int?
  pais_receptor_codigo                         String?              @db.VarChar(255)
  ruc_receptor                                 String?              @db.VarChar(255)
  tipo_contribuyente_receptor                  Int?
  tipo_contribuyente_receptor_desc             String?              @db.VarChar(255)
  tipo_documento_receptor_codigo               Int?
  numero_documento_receptor                    String?              @db.VarChar(255)
  razon_social_receptor                        String?              @db.VarChar(255)
  nombre_fantasia_receptor                     String?              @db.VarChar(255)
  direccion_receptor                           String?              @db.VarChar(255)
  telefono_receptor                            String?              @db.VarChar(255)
  mail_receptor                                String?              @db.VarChar(255)
  numero_orden_compra                          String?              @db.VarChar(255)
  numero_orden_venta                           String?              @db.VarChar(255)
  numero_asiento_contable                      String?              @db.VarChar(255)
  doc_asociado_timbrado_documento_referenciado Int?
  doc_asociado_establecimiento_codigo          Int?
  doc_asociado_punto_expedicion_codigo         Int?
  doc_asociado_numero_documento                Int?
  moneda_codigo                                String?              @db.VarChar(255)
  tipo_cambio                                  Decimal?             @db.Decimal(28, 2)
  subtotal_venta_exento                        Decimal?             @db.Decimal(28, 8)
  subtotal_venta_exoneradas                    Decimal?             @db.Decimal(28, 8)
  subtotal_iva5                                Decimal?             @db.Decimal(28, 8)
  subtotal_iva10                               Decimal?             @db.Decimal(28, 8)
  subtotal_isc                                 Decimal?             @db.Decimal(28, 8)
  total_venta_sin_descuento                    Decimal?             @db.Decimal(28, 8)
  total_descuentos                             Decimal?             @db.Decimal(28, 8)
  descuento_total                              Decimal?             @db.Decimal(28, 8)
  porcentaje_descuento_total                   Decimal              @db.Decimal(28, 0)
  campo_redondeo                               Int
  total_pagar                                  Decimal?             @db.Decimal(28, 8)
  liquidacion_iva5                             Decimal?             @db.Decimal(28, 8)
  liquidacion_iva10                            Decimal?             @db.Decimal(28, 8)
  liquidacion_total_iva                        Decimal?             @db.Decimal(28, 8)
  base_gravada5                                Decimal?             @db.Decimal(28, 8)
  base_gravada10                               Decimal?             @db.Decimal(28, 8)
  total_base_gravada                           Decimal?             @db.Decimal(28, 8)
  total_liquidacion_isc                        Decimal?             @db.Decimal(28, 8)
  total_base_gravada_isc                       Decimal?             @db.Decimal(28, 8)
  monto_anticipo                               Decimal?             @db.Decimal(28, 8)
  monto_total_gs                               Decimal?             @db.Decimal(28, 8)
  cdc                                          String?              @db.VarChar(255)
  procesado                                    Boolean              @default(false, map: "DF__de__procesado__29AC2CE0")
  estado                                       String?              @db.VarChar(255)
  when_created                                 DateTime
  when_modified                                DateTime
  sucursal                                     String               @db.Char(3)
  comprobante                                  String               @db.Char(6)
  num_comp                                     Decimal              @db.Decimal(14, 0)
  tipo_entidad                                 String               @db.Char(2)
  suc_entidad                                  String               @db.Char(3)
  entidad                                      Decimal              @db.Decimal(10, 0)
  condicion_tipo_cambio                        Int?
  condicion_anticipo_codigo                    Int?
  numero_serie                                 String?              @db.VarChar(255)
  codigo_departamento_receptor                 Int?
  codigo_distrito_receptor                     Int?
  codigo_ciudad_receptor                       Int?
  numero_casa_receptor                         Int?
  total_anticipo_por_item                      Decimal?             @db.Decimal(28, 8)
  total_descuento_global_por_item              Decimal?             @db.Decimal(28, 8)
  responsable_tipo_documento_codigo            Int?
  responsable_tipo_documento_descripcion       String?              @db.VarChar(255)
  responsable_numero_documento                 String?              @db.VarChar(255)
  responsable_razon_social                     String?              @db.VarChar(255)
  responsable_cargo                            String?              @db.VarChar(255)
  archivo                                      archivo[]
  autofactura                                  autofactura[]
  documento_asociado                           documento_asociado[]
  estado_historico                             estado_historico[]
  factura                                      factura[]
  items_de                                     items_de[]
  nota_credito                                 nota_credito[]
}

model documento_asociado {
  id                              Decimal   @id(map: "pk_documento_asociado") @db.Decimal(19, 0)
  cdc_documento_referenciado      String?   @db.VarChar(255)
  establecimiento_codigo          Int?
  numero_documento                Decimal?  @db.Decimal(19, 0)
  timbrado_documento_referenciado Decimal?  @db.Decimal(19, 0)
  punto_expedicion_codigo         Int?
  de_id                           Decimal?  @db.Decimal(19, 0)
  tipo_doc_asociado               Int?
  fecha_emision                   DateTime? @db.Date
  tipo_constancia_codigo          Int?
  numero_control_constancia       String?   @db.VarChar(255)
  de                              de?       @relation(fields: [de_id], references: [id], onDelete: NoAction, onUpdate: NoAction, map: "fk_documento_asociado_de_id")

  @@index([de_id], map: "ix_documento_asociado_de_id")
}

model estado_historico {
  id          Decimal  @id(map: "pk_estado_historico") @db.Decimal(19, 0)
  de_id       Decimal  @db.Decimal(19, 0)
  estado      String?  @db.VarChar(255)
  descripcion String?  @db.VarChar(4000)
  tstamp      DateTime
  de          de       @relation(fields: [de_id], references: [id], onUpdate: NoAction, map: "fk_estado_historico_de_id")
}

model evento_cancelacion {
  id                Decimal   @id(map: "pk_evento_cancelacion") @db.Decimal(19, 0)
  id_evento         Decimal   @unique(map: "uq_evento_cancelacion_id_evento") @db.Decimal(10, 0)
  cdc               String    @db.VarChar(255)
  motivo            String    @db.VarChar(255)
  estado            String?   @db.VarChar(255)
  procesado         Boolean   @default(false, map: "DF__evento_ca__proce__2C3393D0")
  mensaje_respuesta String?   @db.VarChar(1000)
  fecha_cancelacion DateTime? @db.DateTime
}

model evento_inutilizacion {
  id                Decimal @id(map: "pk_evento_inutilizacion") @db.Decimal(19, 0)
  id_evento         Decimal @db.Decimal(10, 0)
  motivo            String  @db.VarChar(255)
  numero_timbrado   Int
  establecimiento   Int
  punto_expedicion  Int
  inicio_rango      String? @db.VarChar(255)
  fin_rango         String? @db.VarChar(255)
  tipo_documento    Int
  estado            String? @db.VarChar(255)
  procesado         Boolean @default(false, map: "DF__evento_in__proce__2F10007B")
  mensaje_respuesta String? @db.VarChar(255)
}

model factura {
  id                          Decimal  @id(map: "pk_factura") @db.Decimal(19, 0)
  de_id                       Decimal? @db.Decimal(19, 0)
  condicion_venta             Int?
  monto                       Decimal? @db.Decimal(28, 0)
  condicion_operacion_credito Int?
  numero_cuotas               Int?
  monto_entrega_credito       Decimal? @db.Decimal(28, 0)
  plazo                       String?  @db.VarChar(255)
  modalidad_contratacion      Int?
  entidad_contratacion        Int?
  anho_contratacion           Int?
  secuencia_contratacion      Int?
  fecha_emision_contratacion  String?  @db.VarChar(255)
  modalidad_transporte        Int?
  responsable_emision_nota    Int?
  responsable_pago            Int?
  condicion_entrega           String?  @db.VarChar(255)
  procesado                   Boolean  @default(false, map: "DF__factura__procesa__023D5A04")
  when_created                DateTime
  when_modified               DateTime
  de                          de?      @relation(fields: [de_id], references: [id], onDelete: NoAction, onUpdate: NoAction, map: "fk_factura_de_id")
  cuota                       cuota[]
  pago                        pago[]
}

model items_de {
  id                                    Decimal  @id(map: "pk_items_de") @db.Decimal(19, 0)
  de_id                                 Decimal  @db.Decimal(19, 0)
  codigo_interno_item                   String?  @db.VarChar(255)
  partida_arancelaria_item              Int?
  ncm_item                              Int?
  codigo_general_dncp                   Int?
  codigo_especifico_dncp                Int?
  codigo_producto_gtin                  Decimal? @db.Decimal(19, 0)
  codigo_paquete_gtin                   Decimal? @db.Decimal(19, 0)
  descripcion_item                      String?  @db.VarChar(255)
  unidad_medida                         Int?
  cantidad                              Decimal? @db.Decimal(28, 4)
  informacion_extra_item                String?  @db.VarChar(255)
  precio_unitario_item                  Decimal? @db.Decimal(28, 8)
  descuento_item                        Decimal? @db.Decimal(28, 8)
  valor_total_venta_item                Decimal? @db.Decimal(28, 8)
  afectacion_iva_codigo                 Int?
  proporcion_iva_codigo                 Int?
  tasa_iva_codigo                       Int?
  base_gravada_iva                      Decimal? @db.Decimal(28, 8)
  liquidacion_iva                       Decimal? @db.Decimal(28, 8)
  categoria_codigo_isc                  Int?
  tasa_isc                              Int?
  base_gravada_isc                      Decimal? @db.Decimal(28, 8)
  liquidacion_isc                       Decimal? @db.Decimal(28, 8)
  valor_total_bruto_por_item            Decimal? @db.Decimal(28, 8)
  descuento_global_sobre_precio_item    Decimal? @db.Decimal(28, 8)
  anticipo_particular_sobre_precio_item Decimal? @db.Decimal(28, 8)
  anticipo_global_sobre_precio_item     Decimal? @db.Decimal(28, 8)
  de                                    de       @relation(fields: [de_id], references: [id], onUpdate: NoAction, map: "fk_items_de_de_id")
}

model nota_credito {
  id                         Decimal  @id(map: "pk_nota_credito") @db.Decimal(19, 0)
  de_id                      Decimal? @db.Decimal(19, 0)
  motivo_emision             Int?
  procesado                  Boolean  @default(false, map: "DF__nota_cred__proce__22AA2996")
  when_created               DateTime
  when_modified              DateTime
  codigo_responsable_emision Int?
  de                         de?      @relation(fields: [de_id], references: [id], onDelete: NoAction, onUpdate: NoAction, map: "fk_nota_credito_de_id")
}

model pago {
  id                        Decimal      @id(map: "pk_pago") @db.Decimal(19, 0)
  factura_id                Decimal?     @db.Decimal(19, 0)
  tipo_pago_codigo          Int?
  monto_pago                Decimal      @db.Decimal(28, 0)
  denominacion_tarjeta      Int?
  denominacion_tarjeta_desc String?      @db.VarChar(255)
  ruc_procesadora           String?      @db.VarChar(255)
  forma_pago_tarjeta        Int?
  numero_aprobacion_tarjeta Int?
  numero_cheque             String?      @db.VarChar(255)
  banco_emisor              String?      @db.VarChar(255)
  moneda_codigo             String       @db.VarChar(255)
  tipo_cambio               Decimal?     @db.Decimal(28, 0)
  autofactura_id            Decimal?     @db.Decimal(19, 0)
  autofactura               autofactura? @relation(fields: [autofactura_id], references: [id], onDelete: NoAction, onUpdate: NoAction, map: "fk_pago_autofactura_id")
  factura                   factura?     @relation(fields: [factura_id], references: [id], onDelete: NoAction, onUpdate: NoAction, map: "fk_pago_factura_id")

  @@index([autofactura_id], map: "ix_pago_autofactura_id")
}

model sysdiagrams {
  name         String @db.NVarChar(128)
  principal_id Int
  diagram_id   Int    @id(map: "PK__sysdiagr__C2B05B61164452B1") @default(autoincrement())
  version      Int?
  definition   Bytes?

  @@unique([principal_id, name], map: "UK_principal_name")
}
```

## Prisma Engine Query
```
{"X":{}}
```

**Comment by Jolg42** [comment]
Note: Looks related to an issue with the Decimal native type

**Comment by pimeys** [comment]
Hey. I can see the place where this crashes, but I cannot reproduce the error. The only place where we throw this is when somebody creates a `BigDecimal` with a scale that is bigger than 255, then tries to write that to the SQL Server. The database stores the scale as a 8-bit integer, so we cannot write larger values than 255 here.

Now the code example given seems like it's reading data from the database, but this error only happens on write. For the person fixing this issue, I'd try testing out with some crazy decimals, such as `Decimal(260, 256)` due to scale needing to be smaller than the precision. If this is the case, we should have a validation preventing too large decimal values on SQL Server.

**Comment by Jolg42** [comment]
@agz-jrugel Could you give us an example value that you have in your database that caused this crash?

**Comment by pimeys** [comment]
No need. I found a good number from another issue facing the same problem. This will be fixed in 4.4.0.

**Comment by janpio** [comment]
This was treated as a duplicate of https://github.com/prisma/prisma/issues/15079
